tools/mpremote/mpremote/transport_serial.py

```diff
@@ -571,12 +572,14 @@ def readinto(self, buf):
         return n
 
     def readline(self):
-        l = ''
-        while 1:
-            c = self.read(1)
-            l += c
-            if c == '\\n' or c == '':
-                return l
+        c = self.cmd
+        c.begin(CMD_READLINE)
+        c.wr_s8(self.fd)
+        n = c.rd_u32()
+        buf = bytearray(n)
+        c.rd_bytes(buf)
+        c.end()
+        return bytes(buf)
```

Comment:
This needs to return a str or bytes.  And `rd_bytes` automatically reads the length for you.  I suggest following how read is implemented:
```py
        c = self.cmd
        c.begin(CMD_READLINE)
        c.wr_s8(self.fd)
        data = c.rd_bytes(None)
        c.end()
        if self.is_text:
            data = str(data, 'utf8')
        else:
            data = bytes(data)
        return data
```